src/aleph/vm/controllers/interface.py

```diff
@@ -0,0 +1,91 @@
+import asyncio
+import logging
+from abc import ABC
+from asyncio.subprocess import Process
+from typing import Any, Coroutine, Optional
+
+from aleph_message.models import ItemHash
+from aleph_message.models.execution.environment import MachineResources
+
+from aleph.vm.controllers.firecracker.snapshots import CompressedDiskVolumeSnapshot
+from aleph.vm.network.interfaces import TapInterface
+
+logger = logging.getLogger(__name__)
+
+
+class AlephControllerInterface(ABC):
+    vm_id: int
+    "id in the VMPool, attributed at execution"
+    vm_hash: ItemHash
+    "identifier for the VM definition, linked to an Aleph Message"
+    resources: Any
+    "local resource for the machine"
+    enable_console: bool
+    enable_networking: bool
+    "enable networking for this VM"
+    hardware_resources: MachineResources
+    support_snapshot: bool
+    "Does this controller support snapshotting"
+    guest_api_process: Optional[Process] = None
+    tap_interface: Optional[TapInterface] = None
+    "Network interface used for this VM"
```

Comment:
Shouldn't docstrings be triple quoted?

Reply:
I thought it was triple quote only for functions? Black didn't really complain but I can change if if we want?